docs(react-scheduler): editing options customization (#2716)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/guides/editing.md

## Add a Confirmation Dialog

To enable the confirmation dialog, add the [ConfirmationDialog](../reference/confirmation-dialog.md) plugin. The dialog pops up when a user attempts to delete an appointment or discard edits made to it. If the dialog should not appear in these cases, set the `ignoreDelete` or `ignoreCancel` property to `true`.

.embedded-demo({ "path": "scheduler-editing/delete-confirmation", "showThemeSelector": true })

## Disable one or several editing options

> By default, if you add the [EditingState](../reference/editing-state.md), [AppointmentForm](../reference/appointment-form.md), [IntegratedEditing](../reference/integrated-editing.md) or [EditRecurrenceMenu](../reference/integrated-editing.md),  and [DragDropProvider](../reference/drag-drop-provider.md) plugins, all editing options are turned on. But they can be disabled. Below are the steps to disable them:

Reviewer: Excess paragraph. We should write less but enough to understand how to do something.

- Disable deleting

  To do that, pass `disabled` property to AppointmentForm's [commandButton](../reference/appointment-form.md/#appointmentformcommandbuttonprops) with `deleteButton` ID and use [AppointmentTooltip's](../reference/appointment-tooltip.md) `showDeleteButton` property to  hide its Delete Button.

- Disable updating

  In this case, it's necessary to use [AppointmentForm](../reference/appointment-form.md) in read-only mode. If you want to let users delete appointments, use AppointmentForm's [commandButton](../reference/appointment-form.md/#appointmentformcommandbuttonprops) with `deleteButton` ID and pass `disabled` property to it. If you want to let users create new appointments, do not use read-only mode when a new appointment is being created. If you use [DragDropProvider](../reference/drag-drop-provider.md), consider turning it off. It may be done with the help of `allowDrag` and `allowResize` properties.

> The demo below demonstrates these approaches in action:

Reviewer 2: ```suggestion
The example below demonstrates the described use-cases:
```

> In this case, it's necessary to use [AppointmentForm](../reference/appointment-form.md) in read-only mode. Pass `readOnly={true}` to `AppointmentForm` to disable editing. If you want to let users delete appointments, customize AppointmentForm's [commandButton](../reference/appointment-form.md) with `saveButton` ID and pass `disabled={false}` property to it. If you want to let users create new appointments, do not use read-only mode when a new appointment is being created. If you use [DragDropProvider](../reference/drag-drop-provider.md), consider turning it off. It may be done with the help of `allowDrag` and `allowResize` properties.

Reviewer: I think we should do it like a list instead of a set of sections.
And remove the `appointments` world from names of these sections, it's excess in our context.

### Disable Adding

Since users double-click timetable cells to add appointments, you should remove the double click handler to disable this functionality.

Set the `onDoubleClick` property to `undefined` for the component that renders timetable cells. It is the [timeTableCellComponent](../reference/week-view.md#weekviewtimetablecellprops) for views and the [cellComponent](../reference/all-day-panel.md#alldaypanelcellprops) for the all-day panel.

### Disable Deleting

Users can delete an appointment using the Delete button on the [AppointmentForm](../reference/appointment-form.md) or in the [AppointmentTooltip](../reference/appointment-tooltip.md). Disable this button on the form and hide it from the tooltip if users should not delete appointments.

To disable the Delete button on the `AppointmentForm`, find the [commandButton](../reference/appointment-form.md/#appointmentformcommandbuttonprops) with ID `deleteButton` and set its `disabled` property to `true`.

To hide the Delete button from the `AppointmentTooltip`, disable the `showDeleteButton` property.

### Disable Updating

>   In this case, it's necessary to use [AppointmentForm](../reference/appointment-form.md) in read-only mode. If you want to let users delete appointments, use AppointmentForm's [commandButton](../reference/appointment-form.md/#appointmentformcommandbuttonprops) with `deleteButton` ID and pass `disabled` property to it. If you want to let users create new appointments, do not use read-only mode when a new appointment is being created. If you use [DragDropProvider](../reference/drag-drop-provider.md), consider turning it off. It may be done with the help of `allowDrag` and `allowResize` properties.

Reviewer 2: Please add a link to read-only mode.

```suggestion
Users can use the `AppointmentForm` to update appointments. Switch the `AppointmentForm` to [read-only mode]() to disable this functionality. However, users will not be able to add or delete appointments, too, but these features can be enabled individually.

To enable adding, create a flag that indicates whether an appointment is being added and make the `AppointmentForm` read-only depending on this flag.

To enable deleting, find the [commandButton](../reference/appointment-form.md/#appointmentformcommandbuttonprops) with ID `deleteButton` on the `AppointmentForm` and set its `disabled` property to `false`.

Users can also update appointments via drag-and-drop. Remove the [DragDropProvider](../reference/drag-drop-provider.md) (if you use it) to disable this functionality.
```

## Add a Confirmation Dialog

To enable the confirmation dialog, add the [ConfirmationDialog](../reference/confirmation-dialog.md) plugin. The dialog pops up when a user attempts to delete an appointment or discard edits made to it. If the dialog should not appear in these cases, set the `ignoreDelete` or `ignoreCancel` property to `true`.

.embedded-demo({ "path": "scheduler-editing/delete-confirmation", "showThemeSelector": true })

## Disable Individual Editing Operations

### Read-Only Mode

If users should not edit appointments, import only the [AppointmentForm](../reference/appointment-form.md) plugin and enable its `readOnly` property.

### Disable Adding

> Since users double-click timetable cells to add appointments, you should remove the double click handler to disable this functionality.

Reviewer 3: ```suggestion
Users can double-click timetable cells to add appointments. You should remove the double-click handler to disable this functionality.
```

Set the `onDoubleClick` property to `undefined` for the component that renders timetable cells. It is the [timeTableCellComponent](../reference/week-view.md#weekviewtimetablecellprops) for views and the [cellComponent](../reference/all-day-panel.md#alldaypanelcellprops) for the all-day panel.

### Disable Deleting

Users can delete an appointment using the Delete button on the [AppointmentForm](../reference/appointment-form.md) or in the [AppointmentTooltip](../reference/appointment-tooltip.md). Disable this button on the form and hide it from the tooltip if users should not delete appointments.

To disable the Delete button on the `AppointmentForm`, find the [commandButton](../reference/appointment-form.md/#appointmentformcommandbuttonprops) with ID `deleteButton` and set its `disabled` property to `true`.

To hide the Delete button from the `AppointmentTooltip`, disable the `showDeleteButton` property.

### Disable Updating

> Users can use the `AppointmentForm` to update appointments. Switch the `AppointmentForm` to [read-only mode](../reference/appointment-form.md#properties) to disable this functionality. However, users will not be able to add or delete appointments, too, but these features can be enabled individually.

Reviewer 3: ```suggestion
Users can use the `AppointmentForm` to update appointments. Switch the `AppointmentForm` to [read-only mode](../reference/appointment-form.md#properties) to disable this functionality. However, now users cannot add or delete appointments either, but these features can be enabled individually.
```
